docs/references/plugin-compatibility.md

```diff
@@ -0,0 +1,134 @@
+|key-auth                        |  :white_check_mark:              |  :white_check_mark:                        |
+|kubernetes-sidecar-injector     |  :white_check_mark:              |  :white_check_mark:                        |
+|oauth2                          |  :white_check_mark:              |  :x:                                       |
+|prometheus                      |  :white_check_mark:              |  :white_check_mark:                        |
+|rate-limiting                   |  :white_check_mark:              |  :white_check_mark:                        |
+|request-termination             |  :white_check_mark:              |  :white_check_mark:                        |
+|request-transformer             |  :white_check_mark:              |  :white_check_mark:                        |
+|response-ratelimiting           |  :white_check_mark:              |  :white_check_mark:                        |
+|response-transformer            |  :white_check_mark:              |  :white_check_mark:                        |
+|syslog                          |  :white_check_mark:              |  :white_check_mark:                        |
+|tcp-log                         |  :white_check_mark:              |  :white_check_mark:                        |
+|udp-log                         |  :white_check_mark:              |  :white_check_mark:                        |
+|zipkin                          |  :white_check_mark:              |  :white_check_mark:                        |
+|application-registration        |  :white_check_mark:              |  :x:<sup>1</sup>                           |
+|canary release                  |  :white_check_mark:              |  :white_check_mark:                        |
+|collector                       |  :white_check_mark:              |  :white_check_mark:                        |
+|degraphql                       |  :white_check_mark:              |  :white_check_mark:                        |
+|exit-transformer                |  :white_check_mark:              |  :x:                                       |
+|forward-proxy                   |  :white_check_mark:              |  :white_check_mark:                        |
+|graphql-proxy-cache-advanced    |  :white_check_mark:              |  :white_check_mark:                        |
+|graphql-rate-limiting-advanced  |  :white_check_mark:              |  :white_check_mark:                        |
+|jwt-signer                      |  :white_check_mark:              |  :white_check_mark:                        |
+|kafka-log                       |  :white_check_mark:              |  :white_check_mark:                        |
+|kafka-upstream                  |  :white_check_mark:              |  :white_check_mark:                        |
+|key-auth-enc                    |  :white_check_mark:              |  :x:                                       |
```

Comment:
How about explaining this one as ? This plugin makes sense only in the world where there is a database. In-memory encryption is not a thing in Kong.

Reply:
This one is kind of weird, but I think it is arguably a change in functionality. The difference between key-auth and key-auth-enc is that the persistent store of the token does not contain a plaintext copy of the token, regardless of encryption at the storage layer. key-auth with DB-less does still have a plaintext copy of the token persist somewhere, either a config file or a Kubernetes Secret.

While the current implementation (mutating the admin API key input before persisting it to the database) doesn't make sense in those contexts, alternate implementations are theoretically possible:
* Rewrite the Secret data.
* Have `GET /config` return a version with the ciphertext tokens after you POST the plaintext.

Reply:
Secrets in Kubernetes can be encrypted if someone wants them to be and that's how we explain the encryption at rest story for DB-less mode.
`GET /config` is out of the scope for now.

What you are saying makes sense but it is minor enough for most users that we should still document it for clarity.